rules/ast/ast.go

```diff
@@ -241,12 +241,23 @@ type (
 	// least one of the two operand Nodes must be a VectorNode. The other may be
 	// a VectorNode or ScalarNode. Both criteria are checked at runtime.
 	VectorArithExpr struct {
-		opType BinOpType
-		lhs    Node
-		rhs    Node
+		opType        BinOpType
+		lhs           Node
+		rhs           Node
+		matchType     VectorMatchType
+		matchBy       clientmodel.LabelNames
+		includeLabels clientmodel.LabelNames
 	}
 )
 
+type VectorMatchType int
+
+const (
+	MatchOTO VectorMatchType = iota
+	MatchMTO
```

Comment:
While I think of it, we should error out if someone uses group_left/right with the logical operators as that doesn't make sense.